src-tauri/src/telemetry_manager.rs

```diff
@@ -0,0 +1,321 @@
+                                }
+                            }
+                        }
+                        sleep(timeout);
+                    }
+                } => {},
+                _ = cancellation_token.cancelled() => {
+                    info!("TelemetryManager::start_telemetry_process has been cancelled");
+                }
+            }
+        });
+        Ok(())
+    }
+}
+
+async fn get_telemetry_data(
+    cpu_miner: Arc<RwLock<CpuMiner>>,
+    _gpu_miner: Arc<RwLock<GpuMiner>>,
+    node_manager: NodeManager,
+    config: Arc<RwLock<AppConfig>>,
+    network: Option<Network>,
+) -> Result<TelemetryData, TelemetryManagerError> {
+    let mut cpu_miner = cpu_miner.write().await;
+    let (_sha_hash_rate, randomx_hash_rate, block_reward, block_height, _block_time, is_synced) =
+        node_manager
+            .get_network_hash_rate_and_block_reward()
+            .await
+            .unwrap_or((0, 0, MicroMinotari(0), 0, 0, false));
+    let cpu = match cpu_miner
+        .status(randomx_hash_rate, block_reward)
+        .await
+        .map_err(|e| e.into())
+    {
+        Ok(cpu) => cpu,
+        Err(e) => {
+            warn!(target: LOG_TARGET, "Error getting cpu miner status: {:?}", e);
+            return Err(e);
+        }
+    };
+
+    let hardware_status = HardwareMonitor::current()
+        .write()
+        .await
+        .read_hardware_parameters();
+
+    let config_guard = config.read().await;
+    let is_mining_active = is_synced && cpu.is_mining;
+    let cpu_hash_rate = Some(cpu.hash_rate);
+    let cpu_utilization = hardware_status.cpu.clone().map(|c| c.usage_percentage);
+    let cpu_make = hardware_status.cpu.clone().map(|c| c.label);
+    let gpu_hash_rate = None;
+    let gpu_utilization = hardware_status.gpu.clone().map(|c| c.usage_percentage);
+    let gpu_make = hardware_status.gpu.clone().map(|c| c.label);
+
+    Ok(TelemetryData {
+        app_id: config_guard.anon_id.clone(),
+        block_height,
+        is_mining_active,
+        network: network.map(|n| n.into()),
+        mode: config_guard.mode.clone().into(),
+        cpu_hash_rate,
+        cpu_utilization,
+        cpu_make,
+        gpu_make,
+        gpu_hash_rate,
+        gpu_utilization,
+        resource_used: TelemetryResource::Cpu,
+    })
+}
+
+fn get_airdrop_url() -> String {
+    "https://rwa.y.at".to_string()
+}
+
+async fn send_telemetry_data(data: TelemetryData, airdrop_access_token: Option<String>) {
+    debug!("Telemetry data being sent: {:?}", data);
+
+    let request = reqwest::Client::new();
+    let mut request_builder = request
+        .post(format!("{}/miner/heartbeat", get_airdrop_url()))
+        .json(&data);
+
+    if let Some(token) = airdrop_access_token {
+        request_builder = request_builder.header("Authorization", format!("Bearer {}", token));
+    }
+
+    let result = request_builder
+        .send()
+        .await
+        .map_err(TelemetryManagerError::ReqwestError);
```

Comment:
nit: weird to map the error only to swallow it later

Reply:
True